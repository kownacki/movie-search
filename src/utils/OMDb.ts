import {Movie} from './types';
const OMDB_API_URL = 'https://www.omdbapi.com/';

// Blank posters are described by 'N/A'. Replace them with null.
const nullifyBlankPoster = (movie: Movie): Movie => ({...movie, Poster: movie.Poster === 'N/A' ? null : movie.Poster});

const getMoviesFromResponse = async (response: Response): Promise<Movie[]> => {
  const json = await response.json();
  return (json.Response === "True") ? json.Search.map(nullifyBlankPoster) : [];
};

interface SearchResult {
  movies?: Movie[],
  isSuccess: boolean,
}

export const search = async (query: string): Promise<SearchResult> => {
  // todo add more than 10
  const response = await fetch(`${OMDB_API_URL}?s=${query}&type=movie&apikey=${process.env.REACT_APP_OMDB_API_KEY}`);
  return response.status === 200
    ? {movies: await getMoviesFromResponse(response), isSuccess: true}
    : {isSuccess: false}
};
