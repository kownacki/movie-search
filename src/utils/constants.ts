export const MAX_INPUT_LENGTH = 100;
