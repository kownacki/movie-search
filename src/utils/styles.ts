// Here are kept all common style configuration and utils used across all components.

export const borderRadius = 10;

export const cardBoxShadow = '2px 4px 10px rgba(0, 0, 0, 0.2)';

export const rwdBreakpoints = {
  1: 599,
  2: 419,
};

export const widthAndMinWidth = (width: number): string => `
  width: ${width}px;
  min-width: ${width}px;
`;
