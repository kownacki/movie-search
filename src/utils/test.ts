import * as types from './types';

export const createMovieStub = (changedProps?: Partial<types.Movie>): types.Movie => ({
  imdbID: 'id',
  Poster: 'posterStub',
  Title: 'Title Stub',
  Year: 'yearStub',
  ...changedProps,
});
