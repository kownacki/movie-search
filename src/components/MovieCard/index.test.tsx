import React from 'react';
import {mount} from 'enzyme';
import {MovieCard} from './index';
import * as testUtils from '../../utils/test';

describe('MovieCard', () => {
  it('renders correctly', () => {
    const classNameStub = 'class-name-stub';
    const search = mount(<MovieCard className={classNameStub} movie={testUtils.createMovieStub()}/>);
    expect(search).toMatchSnapshot();
  });
});
