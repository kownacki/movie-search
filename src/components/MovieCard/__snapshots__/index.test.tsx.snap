// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders correctly 1`] = `
<MovieCard
  className="class-name-stub"
  movie={
    Object {
      "Poster": "posterStub",
      "Title": "Title Stub",
      "Year": "yearStub",
      "imdbID": "id",
    }
  }
>
  <div
    className="class-name-stub"
  >
    <styled.img
      alt="Title Stub Poster"
      src="posterStub"
    >
      <img
        alt="Title Stub Poster"
        className="sc-AxjAm hVzkvE"
        src="posterStub"
      />
    </styled.img>
    <Styled(Component)
      title="Title Stub"
      year="yearStub"
    >
      <Component
        className="sc-AxirZ gQSpdr"
        title="Title Stub"
        year="yearStub"
      >
        <div
          className="sc-AxirZ gQSpdr"
        >
          <h2>
            Title Stub
          </h2>
          <p>
            Year of release: 
            yearStub
          </p>
        </div>
      </Component>
    </Styled(Component)>
  </div>
</MovieCard>
`;
